etcdserver/raft.go

```diff
@@ -257,7 +261,9 @@ func (r *raftNode) start(rh *raftReadyHandler) {
 						}
 					}
 					if waitApply {
-						rh.waitForApply()
+						// +1 to sync with scheduled config-change job
+						// if already applied, return immediately without blocking
+						rh.waitForApply(finished + 1)
```

Comment:
Does `waitForApply` need to wait until the entire job is finished or only until the apply waiter is triggered? Instead of expanding the interface to be more dependent on the scheduler details, would `raftDone <- struct{}{}` work here? It'll block until `applyAll` calls `s.applyWait.Trigger`.